pkg/aws/sts.go

```diff
@@ -259,6 +261,15 @@ func (c *awsClient) ListOidcProviders(targetClusterId string) ([]OidcProviderOut
 					return providers, nil
 				}
 			}
+			if idFilter != "" {
```

Comment:
Would it work to supply the config only and check if it's nil instead? One less param